auth-manifest/gen/src/generator.rs

```diff
@@ -0,0 +1,166 @@
+
+        if let Some(priv_keys) = config.vendor_fw_key_info.priv_keys {
+            let sig = self.crypto.ecdsa384_sign(
+                &digest,
+                &priv_keys.ecc_priv_key,
+                &config.vendor_fw_key_info.pub_keys.ecc_pub_key,
+            )?;
+            auth_manifest.preamble.vendor_pub_keys_signatures.ecc_sig = sig;
+
+            let lms_sig = self.crypto.lms_sign(&digest, &priv_keys.lms_priv_key)?;
+            auth_manifest.preamble.vendor_pub_keys_signatures.lms_sig = lms_sig;
+        }
+
+        // Sign the owner manifest keys.
+        if let Some(owner_fw_config) = &config.owner_fw_key_info {
+            if let Some(owner_man_config) = &config.owner_man_key_info {
+                auth_manifest.preamble.owner_pub_keys = owner_man_config.pub_keys;
+
+                let digest = self
+                    .crypto
+                    .sha384_digest(auth_manifest.preamble.owner_pub_keys.as_bytes())?;
+
+                if let Some(owner_fw_priv_keys) = owner_fw_config.priv_keys {
+                    let sig = self.crypto.ecdsa384_sign(
+                        &digest,
+                        &owner_fw_priv_keys.ecc_priv_key,
+                        &owner_fw_config.pub_keys.ecc_pub_key,
+                    )?;
+                    auth_manifest.preamble.owner_pub_keys_signatures.ecc_sig = sig;
+                    let lms_sig = self
+                        .crypto
+                        .lms_sign(&digest, &owner_fw_priv_keys.lms_priv_key)?;
+                    auth_manifest.preamble.owner_pub_keys_signatures.lms_sig = lms_sig;
+                }
+            }
+        }
+
+        // Sign the Image Metadata List.
+        let digest = self
+            .crypto
+            .sha384_digest(auth_manifest.image_metadata_col.as_bytes())?;
+
+        // Sign with the vendor manifest public keys if indicated in the flags.
+        if auth_manifest.preamble.flags & AUTH_MANIFEST_VENDOR_SIGNATURE_REQURIED_FLAG != 0 {
+            if let Some(vendor_man_priv_keys) = config.vendor_man_key_info.priv_keys {
+                let sig = self.crypto.ecdsa384_sign(
+                    &digest,
+                    &vendor_man_priv_keys.ecc_priv_key,
+                    &config.vendor_man_key_info.pub_keys.ecc_pub_key,
+                )?;
+                auth_manifest
+                    .preamble
+                    .vendor_image_metdata_signatures
+                    .ecc_sig = sig;
+
+                let lms_sig = self
+                    .crypto
+                    .lms_sign(&digest, &vendor_man_priv_keys.lms_priv_key)?;
+                auth_manifest
+                    .preamble
+                    .vendor_image_metdata_signatures
+                    .lms_sig = lms_sig;
+            }
+        }
+
+        // Sign with the owner manifest public keys.
+        if let Some(owner_man_config) = &config.owner_man_key_info {
+            if let Some(owner_man_priv_keys) = &owner_man_config.priv_keys {
```

Comment:
Same comment to avoid nested ifs

Reply:
This is slightly different as the owner_man_priv_keys are obtained from owner_man_config. The (Some(), Some()) construct does not work in this case.